main.rb

```diff
@@ -0,0 +1,144 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+  [1, 2, 3].my_each { |n| puts "#{n}!" }
+
+  def my_each_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+    self
+  end
+  [1, 2, 3].my_each_each_with_index { |_n, x| print x }
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = to_a
+    final_array = []
+    arr.my_each { |x| final_array.push(x) if yield(x) }
+    final_array
+  end
+  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].my_select { |x| puts x if x.even? }
+
+  def my_all?(arg = nil)
+    arr = to_a
+    if block_given?
+      arr.my_each { |x| return true unless yield(x) == false }
+      return true
+    elsif arg.nil?
+      return false
+    elsif !arg.nil? && (arg.is_a? Regexp)
+      arr.my_each { |x| return false unless x.match(arg) }
+    elsif (arg.is_a? Class) && !arg.nil?
+      arr.my_each { |x| return false unless [x.class, x.class.superclass].include?(arg) }
+    else
+      arr.my_each { |i| return false if i != arg }
+
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each do |x|
+        return true if yield x
+      end
+
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each do |x|
+        return false unless x.class == arg
+      end
+
+    elsif (arg.class == Regexp) && !arg.nil?
+      my_each do |x|
+        return false unless arg.match(x)
+      end
+
+    else
+      my_each do |x|
+        return true if x == arg
+      end
+
+    end
+    false
+  end
+
+  def my_none?
+    return to_enum unless block_given?
+
+    !my_any?
+  end
+
+  def my_count(args = nil)
+    x = 0
+    if block_given?
+      my_each { |z| x += 1 if yield(z) == true }
+    elsif arg.nil?
+      c = length
+    else
+      my_each { |z| x += 1 if z == args }
+    end
+    c
+  end
```

Comment:
This shows:

```
 18) enumerables #my_count returns the number of items in enum through enumeration
      Failure/Error: expect(array.my_count).to eq array.count
      
      NameError:
        undefined local variable or method `arg' for #<Array:0x0000562406fc4e98>
        Did you mean?  args
      # ./main.rb:93:in `my_count'
      # ./spec/enumerables_spec.rb:342:in `block (3 levels) in <top (required)>'

  19) enumerables #my_count returns the number of items in enum through enumeration::range
      Failure/Error: expect(range.my_count).to eq range.count
      
      NameError:
        undefined local variable or method `arg' for 5..50:Range
        Did you mean?  args
      # ./main.rb:93:in `my_count'
      # ./spec/enumerables_spec.rb:346:in `block (3 levels) in <top (required)>'

  20) enumerables #my_count counts the number of items in enum that are equal to item if an argument is given
      Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)
      
      NameError:
        undefined local variable or method `arg' for #<Array:0x0000562406ffd658>
        Did you mean?  args
      # ./main.rb:93:in `my_count'
      # ./spec/enumerables_spec.rb:350:in `block (3 levels) in <top (required)>'

  21) enumerables #my_count counts the number of elements yielding a true value if a block is given
      Failure/Error: expect(array.my_count(&block)).to eq array.count(&block)
      
        expected: 42
             got: nil
      
        (compared using ==)
      # ./spec/enumerables_spec.rb:354:in `block (3 levels) in <top (required)>'
```

I believe you have a typo on line 93: The correct variable name is `args`, not `arg`.

Also, make your you can produce:

```
array = [1, 2, 3, 4, 5, 6]
block = proc { |num| num < 4 }
array.my_count(&block)
# This should return 3, becase there are 3 numbers (1, 2 and 3) that are lower than 4.
```